### ISSUE TITLE
Incorrect "Your mutator cannot be loaded so default setup has been applied" warning?

### ISSUE THREAD

**Issue by techquestions** [anymaniax/orval#257]
## What are the steps to reproduce this issue?
1. Create custom mutator (in my case: `custom-axios-instance.ts` based on the examples)
2. Add a local import on top of it, e.g.: `import { trackData } from '../../utils/trackData';`
3. Generate the API client

## What happens?

The following warning is shown:

> Your mutator cannot be loaded so default setup has been applied" warning

## What were you expecting to happen?

Looking at the `mutator.ts` code, I am not sure if this warning should be shown:

```
    if (!cached) {
      createLogger().warn(
        chalk.yellow(
          `Your mutator cannot be loaded so default setup has been applied => ${importPath}`,
        ),
      );
    }
```

If my understanding is correct, in this case a file has been correctly loaded and returned but it just hasn't been found in the cache at the time of loading that file. Is that understanding correct?

## What versions are you using?
**Operating System:** macOS 10.15.7
**Package Version:** 6.3.0

**Comment by anymaniax** [comment]
Hello @techquestions, it's normally because orval tried to load the file to check the arguments and generate differently the files based on that. And in your case, there is a problem so orval applies a default config to not crash just for that.

**Comment by techquestions** [comment]
@anymaniax thank you, you are right, it is an issue with one of my files.

Namely, the error I am getting (that causes the warning to show up), is:

```
{
  path: '/Users/user/project/src/api/mutator/custom-axios-instance.ts',
  error: /Users/user/project/node_modules/@scope/library/enums.js:1
  export { Enum1, Enum2 } from './components/Icon/Icon.constants';
  ^^^^^^

  SyntaxError: Unexpected token 'export'
```

So my understanding is that:
1. I have some transitive imports of from modules in `node_modules` in `custom-axios-instance.ts`. These are ES modules.
2. `esbuild` (used in Orval) has problems with bundling files from that module

Do you know what may be the issue here?

**Comment by yekver** [comment]
I have the same issue. In my case some axios config data (ES Module) was imported.

**Comment by anymaniax** [comment]
Hey guys, thanks for this. Can you create an example? Would be easier for me

**Comment by yekver** [comment]
`// orval.config.js`
```javascript
{ ..., override: { mutator: './customAxiosInstance.ts' } }
```

`// customAxiosInstance.ts`
```javascript
import Axios from 'axios'
import config from 'Src/env/config' // this line causes the warning

const customInstance = Axios.create({
  baseURL: config.url
})

export default customInstance
```

**Comment by anymaniax** [comment]
@yekver can I know a bit about your setup to try to handle that case?

**Comment by yekver** [comment]
@anymaniax I've figured out that these two calls are fails parsing:
1. `location.search.includes('paramName')`
2. `localStorage.getItem('paramName')`

**Comment by yekver** [comment]
Should we consider adding a `bug` label to this issue?

Also I'd recommend to change warning to error with message `Failed to parse provided mutator function` because current behaviour is confusing. Mutator wasn't loaded correctly but despite this it's used in a generated code.

**Comment by ninjaoxygen** [comment]
We have a similar use-case, we have runtime config that can be taken from `window.env` or the query string.

Currently accessing anything in `window` in the mutator causes the `Your mutator cannot be loaded so default setup has been applied` warning, for example we have a `window.env.url` that we would like to override `process.env.url` when it is present,

I have tried a lot of workarounds with global var for window etc, but it always leads to the same problem.

**Comment by yekver** [comment]
@anymaniax maybe you can provide files/functions to look at to debug and fix this? I would like to investigate this and make a PR, because this is a blocking issue in my case :)

**Comment by anymaniax** [comment]
I have a solution and currently on it

**Comment by anymaniax** [comment]
@yekver I just did a pull request that should solve the problem

**Comment by yekver** [comment]
> Also I'd recommend to change warning to error with message `Failed to parse provided mutator function` because current behaviour is confusing. Mutator wasn't loaded correctly but despite this it's used in a generated code.

Yeah, great changes, I've already review them. I think we missed this part:

> Also I'd recommend to change the severity from warning to error with message `Failed to parse provided mutator function`. Current behaviour is confusing, mutator wasn't loaded correctly but despite this it's used in a generated code.
